### PR TITLE
Frontend deployment ignores the 'common' utility folder in rules directories

### PR DESCRIPTION
**Pull request by christophe-zurn-sonarsource**

### CODE REVIEW

File: frontend/src/deployment/utils.ts [hunk 1]
```diff
@@ -2,6 +2,9 @@
 import fs from 'fs';
 import path from 'path';
 
+// List of allowed non-language directories used for utility purposes in a rule folder.
+const ALLOWED_UTILITY_DIRECTORY_NAMES = ['common'];
```
Comment (arseniy-sonar) [inline frontend/src/deployment/utils.ts:6]: Should this be a set?
```suggestion
const ALLOWED_UTILITY_DIRECTORY_NAMES = {'common'};
```

**Review by arseniy-sonar — APPROVED** [review]
LGTM, only one minor suggestion, but no need for another round of reveiw

File: frontend/src/deployment/utils.ts (hunk 1 above)
Comment (christophe-zurn-sonarsource) [reply frontend/src/deployment/utils.ts:6]: Unfortunately, there are no set literals in JS/TS. We would need to use `const ALLOWED_UTILITY_DIRECTORY_NAMES = new Set<string>(['common']);`, I'm not sure it's better than simply using arrays.
I don't have a strong opinion on the matter, so if you prefer the alternative with Set, I'll update it accordingly. WDYT ?

Comment (arseniy-sonar) [reply frontend/src/deployment/utils.ts:6]: Ah, that's a pity. I think simplicity is more important here, so let's keep an array

**Pull request merged**